CMakeLists.txt

```diff
@@ -13,6 +13,11 @@ find_package(projectM4 REQUIRED)
 find_package(GStreamer REQUIRED COMPONENTS gstreamer-audio gstreamer-gl gstreamer-pbutils gstreamer-video)
 find_package(GLIB2 REQUIRED)
 
+if(projectM4_VERSION VERSION_LESS 4.2.0)
```

Comment:
You could have CMake check this directly in the find_package() command using this (rather weird) syntax:
```cmake
find_package(projectM4 4.2.0...< REQUIRED)
```

Reply:
It doesn't seem to like that 
CMake Error at CMakeLists.txt:12 (find_package):
  find_package called with invalid argument "4.1.0...<"

The docs aren't super helpful to me https://cmake.org/cmake/help/latest/command/find_package.html#find-package-version-format

Reply:
Yeah, the docs are relatively vague. I guess the upper limit can't be excluded, which I should have seen. This syntax here is probably the right one, it basically says "minimum 4.2.0, and everything before 5.0".

```cmake
find_package(projectM4 4.2.0...<5.0.0 REQUIRED)
```

Reply:
really, you have to specify the upper limit? that's fucked

Reply:
Inconvenient for sure. You could in theory just put `9999` there, but I'd also prefer a notation like `>=4.2.0` or even a comma/semicolon-separated list of ranges so you may exclude certain versions, e.g. bugged ones, like `>=1.0.0...<1.5.0,>=1.5.3` to exclude versions from 1.5.0 up to 1.5.2 if those have a bug that was fixed in 1.5.3. But it is what it is. Putting 5.0.0 there should suffice, as the projectM-4 package will never reach this version.